roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,698 @@
+        except OpenShiftCheckException as exc:
+            self.register_failure(exc)
+        else:
+            if self.want_full_results:
+                self.register_file('node-config.yaml', node_conf_contents)
+            node_conf = yaml.load(node_conf_contents)
+            node_name = node_conf.get('nodeName')
+
+        return node_name
+
+    def process_pods_for_node(self, node_name):
+        """Iterate over pods, and for each one, add the pid from the first
+        container and the MAC address and Open vSwitch bridge port and VNID
+        corresponding to the pod's netns."""
+        pod_addr_to_port, pod_addr_to_vnid = self.get_flow_maps()
+        for pod in self.get_resource('pods'):
+            # Skip the pod if it is not on this node.
+            if pod['spec'].get('nodeName') != node_name:
+                continue
+
+            # Skip the pod if it does not have a ready container or if it is
+            # on the host's network.
+            status = pod['status']
+            if 'containerStatuses' not in status or \
+               status['containerStatuses'][0].get('ready') is not True or \
+               'hostNetwork' in pod['spec']:
+                continue
+
+            # Get the pid of a container in the pod; we will need the pid in
+            # order to enter the container's netns.
+            cid = status['containerStatuses'][0]['containerID'] \
+                .replace('docker://', '')
+            try:
+                pid = self.read_command_output(['docker', 'inspect',
+                                                '-f', '{{.State.Pid}}', cid])
+                if not pid or pid == '0':
+                    raise OpenShiftCheckException(
+                        'CouldNotReadPodPid',
+                        'Could not determine pid of pod %s in namespace %s' %
+                        (pod['metadata']['name'],
+                         pod['metadata']['namespace']))
+                pod['pid'] = pid
+            except OpenShiftCheckException as exc:
+                self.register_failure(exc)
+                continue
+
+            # Look up and store the MAC address, OVS bridge port and VNID for
+            # the pod so that the connectivity checks can easily look them up
+            # later.
+            port = pod_addr_to_port.get(status['podIP'], None)
+            if not port:
+                self.register_failure('Could not find port for %s!' %
+                                      status['podIP'])
+                continue
+            pod['ovs_bridge_port'] = port
+
+            vnid = pod_addr_to_vnid.get(status['podIP'], '0')
+            if not vnid:
+                self.register_failure('Could not find VNID for %s!' %
+                                      status['podIP'])
+                continue
+            pod['vnid'] = vnid
+
+            # XXX: Would it be worthwhile to create an nsenter module?
```

Comment:
Probably, but @sdodson would be the better person to ask.

I'd say it's a TODO.

Reply:
Changed to a TODO.